### DISCUSSION TITLE
Is there anyway to add permission for limit admin rest api access or custom message via docker compose mode

### DISCUSSION THREAD

**Discussion by leapit** [ory/hydra#2739] [Q&A] [answered]
**The first question**
Run hydra with docker compose `quickstart.yml`,According the doc,add client via REST api,like this
```
curl -X POST 'http://localhost:4445/clients' \
-H 'Content-Type: application/json' \
--data-raw '{
    "client_id": "myidentityapp",
    "client_name": "MyIdentityPpp",
    "client_secret": "mysecret",
    "grant_types": ["authorization_code", "refresh_token"],
    "redirect_uris": ["http://localhost:3000/callbacks"],
    "response_types": ["code", "id_token"],
    "scope": "offline users.write users.read users.edit users.delete",
    "token_endpoint_auth_method": "client_secret_post"
}'
```
But in production mode,generally we do't want to expose 4445 to outside,otherwise any one can create a client remotely,is there a way to add a permission for this?

**The second question**
I production mode,when you access port 4444 or port 4445,will show below message,
```
{"error": "Error 404 - The requested route does not exist. Make sure you are using the right path, domain, and port."}
```

Is there any way can custom this message by modifying docker-compose.yml?

**Comment by vinckr** [comment, marked as answer]
> Is there a way to add a permission for this

Sorry, but what do you want to add a permission to? To access port 4445? That you need to configure in your API gateway or load balancer. Check out[ Ory Oathkeeper](https://www.ory.sh/oathkeeper/docs/next/) for example.
For some guidance on how to move to production, please refer to this document: https://www.ory.sh/hydra/docs/production. 

> Is there any way can custom this message by modifying docker-compose.yml?

No, but you can display a different message in your front-end UI.
